app/src/main/java/com/android/streetworkapp/ui/park/AddEvent.kt

```diff
@@ -0,0 +1,305 @@
+  val minParticipants = 2 // We need at least 2 users for every event
+  val maxParticipants = 10 // An event can not have more then 10 participants
+  var sliderPosition by remember { mutableFloatStateOf(minParticipants.toFloat()) }
+
+  Column(
+      verticalArrangement = Arrangement.Center,
+      horizontalAlignment = Alignment.CenterHorizontally,
+      modifier = Modifier.fillMaxWidth()) {
+        Slider(
+            modifier = Modifier.fillMaxWidth(0.8f).testTag("sliderMaxParticipants"),
+            value = sliderPosition,
+            onValueChange = {
+              sliderPosition = it
+              event.maxParticipants = sliderPosition.toInt()
+            },
+            colors =
+                SliderDefaults.colors(
+                    thumbColor = MaterialTheme.colorScheme.secondary,
+                    activeTrackColor = MaterialTheme.colorScheme.secondary,
+                    inactiveTrackColor = MaterialTheme.colorScheme.secondaryContainer,
+                ),
+            steps = 7,
+            valueRange = minParticipants.toFloat()..maxParticipants.toFloat())
+        Text(text = sliderPosition.toInt().toString())
+      }
+}
+
+/**
+ * Used to change the date and the hour of the event
+ *
+ * @param event the event that will be updated
+ */
+@ExperimentalMaterial3Api
+@Composable
+fun TimeSelection(event: Event) {
+  var showDatePicker by remember { mutableStateOf(false) }
+  var showTimePicker by remember { mutableStateOf(false) }
+
+  val currentTime = Calendar.getInstance()
+  val datePickerState = rememberDatePickerState(currentTime.timeInMillis)
+  val timePickerState =
+      rememberTimePickerState(
+          initialHour = currentTime.get(Calendar.HOUR_OF_DAY),
+          initialMinute = currentTime.get(Calendar.MINUTE),
+          is24Hour = false,
+      )
+
+  val currentTimeSelection =
+      datePickerState.selectedDateMillis?.plus(
+          (TimeUnit.HOURS.toMillis(timePickerState.hour.toLong()) +
+              TimeUnit.MINUTES.toMillis(timePickerState.minute.toLong())))
+
+  val selectedDate =
+      datePickerState.selectedDateMillis?.let { convertMillisToDate(currentTimeSelection!!) } ?: ""
+
+  Box(modifier = Modifier.fillMaxWidth(0.9f)) {
+    OutlinedTextField(
+        value = selectedDate,
+        onValueChange = {},
+        label = { Text("When do you want to train?") },
+        readOnly = true,
+        trailingIcon = {
+          Row {
+            IconButton(
+                modifier = Modifier.testTag("dateIcon"),
+                onClick = { showDatePicker = !showDatePicker }) {
+                  Icon(imageVector = Icons.Default.DateRange, contentDescription = "Select date")
+                }
+            IconButton(
+                modifier = Modifier.testTag("timeIcon"),
+                onClick = { showTimePicker = !showTimePicker }) {
+                  Icon(imageVector = Icons.Default.AddCircle, contentDescription = "Select time")
```

Comment:
Suggestion: I think this icon is a bit misleading, I couldn't tell it was for the time at first glance. Should probably be changed to a clock or something similar to remove ambiguities

Reply:
I agree with you. There was no clock icon in the basic set of icon so I had to import "libs.androidx.material.icons.extended". Thank you!